Cli options

romzie opened the pull request:
The image -> json palette works fine on cli, but the implementation behind produces an incompatible json file (old version, e.g. without `_background` attribute). Needs to be fixed later on.

nyancol committed:
[config] Adding a config file

nyancol committed:
[hyperplan_creator] Adding a function to calibrate hyperplans

It is now possible to recalibrate an existing base hyperplan

nyancol committed:
[test] Adding unit tests

nyancol committed:
[pywall support] ColorExtractor returns a dictionary of colors

nyancol committed:
[pywall] Returns 16 colors in addition of bg and fg

If necessary, the last color is repeated until there are 16 colors

nyancol committed:
[export] Exporting vim and iTerm

nyancol committed:
[style] Imports corrected

rvdz committed:
Better hyperplan calibration + better color gen

nyancol committed:
[export] iTerm and Vim support

rvdz committed:
getColors can be now be launched anywhere

nyancol committed:
[structure] Structuring project

nyancol committed:
[config] Added config initialization

nyancol committed:
[setup] Adding a project setup

nyancol committed:
[test] Adding configuration tests

nyancol committed:
[config] Small fixes after merge and comments

Commit:
Added script to create a profile for gnome terminal

First version: colors hardcoded for test purpose
TODO: colors in arguments

nyancol committed:
[config] corrected config

nyancol committed:
[manifest] Adding a MANIFEST

nyancol committed:
[setup] Now supports 'install' command

nyancol committed:
[python3.5] Fix enum.Flag -> enum.Enum

Commit:
Added arguments input

./export_gnome.sh -h to display usage

Commit:
Added arguments input

./export_gnome.sh -h to display usage

Commit:
Export Gnome + Yabar

Added config parameters for both //not working
Added template for yabar.conf //prototype
Not tested yet //running error

nyancol committed:
[gh-pages] Adding HTML page

nyancol committed:
[iterm] Added "set as default" option

nyancol committed:
[lightline] Fix when only one color in reused label

nyancol committed:
[iterm] Fix (label - ansi index) pairing

rvdz committed:
Merge pull request #6 from rvdz/docs

Added HTML web site

nyancol committed:
[setup] Adding dependencies

nyancol committed:
[config] A custom configuration file can be used

nyancol committed:
[doc] Adding filters and targets documentation

nyancol committed:
[config] Adding method to reconfigure targets

romzie committed:
Merge branch 'master' of github.com:rvdz/hapycolor into export_gnome

romzie committed:
[EXPORT] Gnome

Added config management
Fixed export of non-bold colors
Working fine on Debian 9 with gnome-terminal 3.22

romzie committed:
[Export] Gnome

Fixed config path

romzie committed:
Merge branch 'master' into export_gnome

nyancol committed:
[imgur] Adding imgur support

nyancol committed:
[iterm] Added support to encoded property lists

nyancol committed:
[input image] Supporting png with alpha and greyscale formats

romzie committed:
[export] Gnome: pulled master modifications

Removed forgotten debug print

romzie committed:
[Yabar] Export v1

Method: use tokens in yabar config file,
        then hapycolor will replace them by
        color values

romzie committed:
[yabar] export

Added new color token defined_color
Fixed random token

romzie committed:
[export] Gnome Terminal: merge PR

Added README explanations
Improved code to handle exception
Fixed config file

romzie committed:
Merge pull request #7 from rvdz/export_gnome

Export palette for Gnome terminal

nyancol committed:
[style] Style fixes

romzie committed:
[export] Yabar

Rewrite token for defined colors
Added config management

romzie committed:
[export] Yabar: style edit

romzie committed:
[export] Yabar

Minor fixes and style

romzie committed:
[doc] README for targets use

Added Yabar export explanation and guide

nyancol committed:
[filters] Major changes on the Luminosity filter

- Added support to the incoming hyperplan editor
- Saturation filtering rewritten
- Luminosities filtering rewritten

nyancol committed:
[reduction] Added pre-filtering optimization

romzie committed:
[export] yabar: now creating a new conf file

romzie committed:
Merge pull request #10 from rvdz/yabar

Export palette in Yabar config

nyancol committed:
[fix] Fixed hyperplan's paths

romzie committed:
[export] [doc] wallpaper linux

Added README entry for Linux wallpaper
Edited wallpaper target to add Linux compatibility
FIXME: test gsettings

romzie committed:
[options] command line

Added reconfigure option
Added print of target lists

romzie committed:
[export] [style] PR changes requested

Edited the configuration key for more relevance
Edited os handling using config methods

romzie committed:
Merge pull request #11 from rvdz/wallpaper_linux

Wallpaper compatibility with Linux

romzie committed:
[options] enable/disable target

Reworked and added some functions in targets/__init__.py
Added enable and disable feature in cli options
It may be possible to factorize those two options...

romzie committed:
[options] [style] export only

Created mutually exclusive group
Added export only option
Minor style edits

romzie committed:
[options] argparse -> docopt

Switched from argparse to docopt for obvious reasons

nyancol reviewed (commented):
Nice work!

nyancol committed:
[vim] Adding unique groups support

nyancol committed:
[options] Fixing a totally harmless bug

nyancol committed:
[wallpaper] Fixing configuration key error

nyancol committed:
[vim] Automatic plugin search

nyancol committed:
[comments] Adding comments in Reducer and LumFilter

nyancol committed:
[filters] Factorizing analyze functions + comments added

nyancol committed:
[style] Improved coding style with `pycodestyle`

nyancol committed:
[trolling] Adding a new feature for b&w images

romzie committed:
[options] config is now printable

Improved esthetics of messages.
Added --print-config option

nyancol reviewed (commented):
Just found some more issues

nyancol reviewed (commented):
Found an issue

nyancol committed:
[options] Small fix

romzie committed:
[options] uninitialized targets

View config prints message in case target is not initialized.
Option --dir has no longer a short alternate.
Short option -d has been assigned to --disable.
Fixed a few things mentionned in PR messages.

nyancol committed:
[options] Adding imgur support

nyancol committed:
[options] Adding json support

nyancol committed:
[options] Debuging and small fixes

nyancol committed:
[doc] Moving doc

nyancol committed:
Merge branch 'master' into cli_options

nyancol committed:
Merge branch 'master' into cli_options
